adrs/0541531435/README.md

```diff
@@ -0,0 +1,118 @@
+the project. Or in cases that you would like to run the project in production.
+
+Having a directory that reflects such information is crucial to mitigate mistakes
+in application deployments, or misuse of libraries.
+
+### About test frameworks
+
+Sometimes you may have multiple test frameworks for your project, most likely
+for `end-to-end`, `integration`, or `unit` tests, but not limited to those
+concepts. Therefore, people have tried to come up with `./tests/integration`,
+`./tests/unit`, `./tests/end-to-end`.
+
+Although that may be a clean way to structure your tests, the problems arrive
+when you try to articulate the differences between those categories without
+getting into an in-depth conversation about X vs. Y and end up with unclear
+boundaries, relying too much on the programmer's experiences to articulate the
+most optimal definition or decision.
+
+Additionally, if you try to migrate your project to another testing tool that
+falls into one of those possible categories, you will mingle tests that are
+run differently depending on the underline tool, and that may be really
+important in some cases.
+
+### About documentation
+
+Documentation tools are amazingly diverse in the ecosystem; such tools require
+different configurations and code structure, therefore mingle tools may cause
+incompatible situations, or confusing documentation source code.
+
+Also, it is unclear today what should I do if I have multiple documentation
+websites (internal or public as an example), and how to manage them.
+
+## Resolution
+
+* You **must** follow the following directory structures:
+
+  ```text
+  .
+  ├── package.json
+  ├── bin
+  ├── dist
+  ├── docs
+  │   ├── storybook
+  │   └── website
+  │   └── website-next
+  │   └── [another website or tool]
+  ├── examples
+  ├── lib
+  ├── src
+  └── tests
+      ├── cypress
+      └── jest
+      └── puppeteer
+      └── [another testing tool]
+  ```
+
+* In the `./tests` directory, you **must** use the tool's name to reflect the
+  tool used to run the tests.
+* You **must** place `storybook` documentation under `./docs`.
+* In the `./docs` directory, you **must** separate documentation based on the
+  underline tool or context (for example, **internal** or **public**).
+* `./src` **must** required compilation step, and you **must** compile the
+  source code into `./dist` directory.
+* `./dist` **must** required to compile `./src` directory to generate
+  the source code, or any other compilation step in required cases.
+
+### About files and directories
+
+* **./package.json:** the NodeJS package configuration.
+* **./bin:** contains executables files.
+* **./dist:** contains compiled source code.
+* **./docs:** contains the documentation of the project.
+* **./examples:** contains examples of the project's usage, primarily used
+  in library projects.
+* **./lib:** source code that does not require compilation.
+* **./src:** source code that requires some compilation.
```

Comment:
There might be a missed opportunity here to make a recommendation FOR or AGAINST expanding projects into multi-package projects as they grow -- like yarn workspaces.

Reply:
That is outside of the scope of this ADR since workspaces are just multiple projects/packages.

Reply:
Or to put this another way.  If I identify something that could/should be a stand-alone module, where does that go?

Reply:
Depends on your repository (notice that I didn't say NodeJS project), if you have a mono repo, then depends on how you setup the mono repo.

This is for the project/package itself, maybe we can write up something for mono repo?

Reply:
The mono repo would open interesting questions.